### ISSUE TITLE
[Question]

### ISSUE THREAD

**Issue by Athena-ya** [callous-youth/BOAT#81]
### Required prerequisites

- [x] I have read the documentation <https://boat.readthedocs.io/en/main/>.
- [x] I have searched the [Issue Tracker](https://github.com/callous-youth/BOAT/issues) to make sure this question has not already been answered.

### 标题
"Clarification on gradient order execution when multiple gm_op/na_op are specified"

#### Environment
- OS: macOS
- Python: 3.11.13
- torch: 2.0+
- boat-torch: latest from git
- higher: 0.2.x

#### What I'm trying to do
I'm experimenting with data hyper-cleaning using various combinations of gradient operators (gm_op) and nested adjoint operators (na_op). I want to understand how BOAT determines the actual execution order of these operators.

#### Questions
1. When I specify `gm_op=["DI", "NGD"]` and `na_op=["CG"]`, is the order preserved as specified, or does BOAT's internal sorting (e.g., in `boat_torch/boat_opt.py`) override it?

2. The `HyperGradientRules.set_gradient_order()` and `DynamicalSystemRules.set_gradient_order()` methods define global ordering rules. How do these interact with the operator order specified in `boat_config`?

3. Should users manually set these Rules before calling `build_ll_solver()`, or is there a recommended default ordering?

4. Is there a documented "best practice" for ordering operators to achieve faster convergence on specific problem types?

#### What I tried
```python
from boat_torch.utils import HyperGradientRules, DynamicalSystemRules
import boat_torch as boat

# Setting up ordering rules
HyperGradientRules.set_gradient_order([
    ["PTT", "RGT", "FOA"],
    ["IAD", "RAD", "FD", "IGA"],
    ["CG", "NS"],
])
DynamicalSystemRules.set_gradient_order([
    ["GDA", "DI"],
    ["NGD", "DM"],
])

# Then configuring with specific gm_op/na_op order
boat_config["gm_op"] = ["DI", "NGD"]
boat_config["na_op"] = ["CG"]
```

Running the same experiment with different orderings (e.g., `["NGD", "DI"]`) produces different convergence curves, but it's unclear which factor drives the difference: the Rules ordering or the config list ordering.

#### Suggestion
It would be helpful if the documentation or API docstrings clarified:
1. The precedence: does `boat_config` order override `Rules.set_gradient_order()`?
2. Whether the global Rules are meant to be set once per session or per problem instance
3. An example showing the relationship between `boat_config["gm_op"]` ordering and actual computational flow

This clarification would significantly improve the onboarding experience for new users trying to customize optimizer strategies.
